ethicml/evaluators/parallelism.py

```diff
@@ -76,22 +81,27 @@ def arrange_in_parallel(
     runner = Parallel(n_jobs=num_jobs, verbose=10, backend="loky")
     assert len(algos) >= 1
     assert len(data) >= 1
+    assert len(seeds) == len(data)
     assert isinstance(data[0][0], DataTuple)
     assert isinstance(data[0][1], TestTuple)
     # ================================== create queue of tasks ====================================
     # for each algorithm, first loop over all available datasets and then go on to the next algo
-    results = runner(_run(algo, data_item) for algo in algos for data_item in data)
+    results = runner(
+        _run(algo, data_item, seed) for algo in algos for (data_item, seed) in zip(data, seeds)
+    )
     # return [[result_dict[(i, j)] for j in range(len(data))] for i in range(len(algos))]
     # we have to reconstruct the nested list from the flattened list
     return [[results[i * len(data) + j] for j in range(len(data))] for i in range(len(algos))]
 
 
 @delayed
-def _run(algo: Algorithm[_RT], train_test_pair: TrainTestPair) -> _RT:
+def _run(algo: Algorithm[_RT], train_test_pair: TrainTestPair, seed: int) -> _RT:
     train, test = train_test_pair
     # do the work
     try:
         result: _RT = algo.run(train, test)
+        if isinstance(result, Prediction):
+            result.info["model_seed"] = seed
     except RuntimeError:
         result = Prediction(hard=pd.Series([np.NaN] * len(test)))
     return result
```

Comment:
Just realised that this will not work in the case that a PreProcess method fails.

Reply:
I'll add an issue as it's a pretty rare event

Reply:
Right. Doesn't mypy complain?

Reply:
no. I don't think that it can know about this.